### PR TITLE
WIP [GEOS-11000] WPS process to provide elevation profile for a linestring

### PR DESCRIPTION
**Pull request by Mike7311**
[GEOS-11000](https://osgeo-org.atlassian.net/browse/GEOS-11000)

[[GEOS-11000] WPS process to provide elevation profile for a linestring](https://osgeo-org.atlassian.net/browse/GEOS-11000)
Addition of new WPS process, which provides elevation profile for specified linestring as a result.
Also can reprojects result to different crs and adjust elevation using another layer.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by Mike7311** [comment]
Need permission on the geoserver-developer list. Until then WIP

File: src/community/wps-longitudinal-profile/src/main/java/org/geoserver/wps/longitudinal/OperationInfo.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
+package org.geoserver.wps.longitudinal;
```
Comment (ianturton) [inline src/community/wps-longitudinal-profile/src/main/java/org/geoserver/wps/longitudinal/OperationInfo.java:1]: missing header

File: src/community/wps-longitudinal-profile/src/main/java/org/geoserver/wps/longitudinal/ProfileInfo.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package org.geoserver.wps.longitudinal;
```
Comment (ianturton) [inline src/community/wps-longitudinal-profile/src/main/java/org/geoserver/wps/longitudinal/ProfileInfo.java:1]: missing header

File: src/community/wps-longitudinal-profile/src/test/java/org/geoserver/wps/longitudinal/test/LongitudinalProfileProcessTest.java [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+            "<wps:ResponseForm>\n"
+                    + "    <wps:RawDataOutput mimeType=\"application/json\">\n"
+                    + "      <ows:Identifier>result</ows:Identifier>\n"
+                    + "    </wps:RawDataOutput>\n"
+                    + "  </wps:ResponseForm>\n";
+    private static final String LINESTRING =
+            "<wps:Input>\n"
+                    + "  <ows:Identifier>linestringWkt</ows:Identifier>\n"
+                    + "      <wps:Data>\n"
+                    + "        <wps:ComplexData mimeType=\"text/xml; subtype=gml/3.1.1\">"
```
Comment (ianturton) [inline src/community/wps-longitudinal-profile/src/test/java/org/geoserver/wps/longitudinal/test/LongitudinalProfileProcessTest.java:42]: that is WKT not GML

**Review by ianturton — COMMENTED** [review]
Other than those minor points I'm happy

File: doc/en/user/source/extensions/wps-longitudinal-profile/index.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
+.. _wpslongitudinal:
+
+WPS longitudinal profile process
+===================
```
Comment (mprins) [inline doc/en/user/source/extensions/wps-longitudinal-profile/index.rst:4]: the underline is too short

File: doc/en/user/source/extensions/wps-longitudinal-profile/index.rst [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+WPS longitudinal profile process provides ability to calculate elevation profile for linestring
+* Reproject result to different CRS
+* Adjust elevation profile based on additional layer
+
+Installing the WPS longitudinal profile process
+-------------------------------------
```
Comment (mprins) [inline doc/en/user/source/extensions/wps-longitudinal-profile/index.rst:11]: the underline is too short

**Review by mprins — CHANGES_REQUESTED** [review]

File: src/community/wps-longitudinal-profile/src/main/java/org/geoserver/wps/longitudinal/OperationInfo.java (hunk 1 above)
Comment (Mike7311) [reply src/community/wps-longitudinal-profile/src/main/java/org/geoserver/wps/longitudinal/OperationInfo.java:5]: fixed

File: doc/en/user/source/extensions/wps-longitudinal-profile/index.rst (hunk 4 above)
Comment (Mike7311) [reply doc/en/user/source/extensions/wps-longitudinal-profile/index.rst]: fixed

**Comment by dromagnoli** [comment]
Hi @Mike7311.
Please, also make sure to create an ext-XXX file on community\release so that we will get the associated zip generated in the nightly downloads. See the other modules present here. they are ext-yourmodulename.xml containing <id>yourmodulename-plugin</id> and the list of needed files to be packaged in the zip.

File: src/community/release/ext-wps-longitudinal-profile.xml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<assembly>
+  <id>wps-longitudinal-profile</id>
```
Comment (dromagnoli) [inline src/community/release/ext-wps-longitudinal-profile.xml:2]: make sure to append -plugin to the id:
<id>wps-longitudinal-profile-plugin</id>

If my memory serves me right, in the past I have defined a new module using an id without the -plugin suffix and it was not showing up in the download list.

Comment (Mike7311) [reply src/community/release/ext-wps-longitudinal-profile.xml]: done

File: doc/en/user/source/extensions/wps-longitudinal-profile/index.rst [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+If projection parameter is provided, points of profile will be reprojected to target CRS, otherwise to CRS
+of provided Ewkt linestring.
+
+Response contains following objects:
+profile - contains array of points of the profile
+infos - general info on process result
+
+profile object contains array of points
+each point has following values:
+distance - distance to this point from the beginning of the profile (first point) in units of CRS
```
Comment (MV88) [inline doc/en/user/source/extensions/wps-longitudinal-profile/index.rst:54]: this should be called "totalDistanceToThisPoint" right?

Comment (Mike7311) [reply doc/en/user/source/extensions/wps-longitudinal-profile/index.rst]: fixed

**Comment by aaime** [comment]
The backport to `2.21.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply 38c7db3a28... [GEOS-11000] WPS process to provide elevation profile for a linestring
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".

```
</details>

<details>
<summary>stdout</summary>

```
Auto-merging src/community/pom.xml
CONFLICT (content): Merge conflict in src/community/pom.xml

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-2.21.x 2.21.x
# Navigate to the new working tree
cd .worktrees/backport-2.21.x
# Create a new branch
git switch --create backport-6868-to-2.21.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 38c7db3a28f6491c51d6bc7f14da4aead6e6684e,16f200ab5799f85edfa43e423c96448445ce020d,9c05177d70b40a32d7815df70d66d68a037b087e,b217a15bad5436acdf9aa5adfa08f37b01505808,84ca4cd0827e38853f596172bbd56c7c7488ef96,e92464513bf747b84407d2f5499d701eaaa47a24
# Push it to GitHub
git push --set-upstream origin backport-6868-to-2.21.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-2.21.x
```
Then, create a pull request where the `base` branch is `2.21.x` and the `compare`/`head` branch is `backport-6868-to-2.21.x`.

**Comment by aaime** [comment]
The backport to `2.22.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply 38c7db3a28... [GEOS-11000] WPS process to provide elevation profile for a linestring
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".

```
</details>

<details>
<summary>stdout</summary>

```
Auto-merging src/community/pom.xml
CONFLICT (content): Merge conflict in src/community/pom.xml

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-2.22.x 2.22.x
# Navigate to the new working tree
cd .worktrees/backport-2.22.x
# Create a new branch
git switch --create backport-6868-to-2.22.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 38c7db3a28f6491c51d6bc7f14da4aead6e6684e,16f200ab5799f85edfa43e423c96448445ce020d,9c05177d70b40a32d7815df70d66d68a037b087e,b217a15bad5436acdf9aa5adfa08f37b01505808,84ca4cd0827e38853f596172bbd56c7c7488ef96,e92464513bf747b84407d2f5499d701eaaa47a24
# Push it to GitHub
git push --set-upstream origin backport-6868-to-2.22.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-2.22.x
```
Then, create a pull request where the `base` branch is `2.22.x` and the `compare`/`head` branch is `backport-6868-to-2.22.x`.

**Comment by aaime** [comment]
The backport to `2.23.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply 38c7db3a28... [GEOS-11000] WPS process to provide elevation profile for a linestring
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".

```
</details>

<details>
<summary>stdout</summary>

```
Auto-merging src/community/pom.xml
CONFLICT (content): Merge conflict in src/community/pom.xml

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-2.23.x 2.23.x
# Navigate to the new working tree
cd .worktrees/backport-2.23.x
# Create a new branch
git switch --create backport-6868-to-2.23.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 38c7db3a28f6491c51d6bc7f14da4aead6e6684e,16f200ab5799f85edfa43e423c96448445ce020d,9c05177d70b40a32d7815df70d66d68a037b087e,b217a15bad5436acdf9aa5adfa08f37b01505808,84ca4cd0827e38853f596172bbd56c7c7488ef96,e92464513bf747b84407d2f5499d701eaaa47a24
# Push it to GitHub
git push --set-upstream origin backport-6868-to-2.23.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-2.23.x
```
Then, create a pull request where the `base` branch is `2.23.x` and the `compare`/`head` branch is `backport-6868-to-2.23.x`.

**Pull request merged**